apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/JsonElement.kt

```diff
@@ -0,0 +1,61 @@
+package com.apollographql.apollo.api
+
+/**
+ * A wrapper class for representation of custom GraphQL type value, used in user provided [CustomScalarAdapter]
+ * encoding / decoding functions.
+ **/
+sealed class JsonElement {
```

Comment:
nit: should we move this into `/json` subfolder?

Reply:
Good question. We currently have `com.apollographql.apollo.api.internal.json` where the Json parser (`BufferedSourceJsonReader`) resides. 

Technically, we could make the json parser work with `JsonElement` but that will introduce some unnecessary wrapping and I'm afraid it'll only slow down parsing... 
So `JsonElement` is right now only used to wrap Kotlin instances into more type-safe/user-friendly types. In that sense I feel that it should be closer to `CustomScalarAdapter` than `BufferedSourceJsonReader`, even if it's about Json. Also the `com.apollographql.apollo.api` package isn't too much cluttered right now. It contains ~30 classes, some of which can maybe be moved to `internal`. All that makes me want to keep it in the root package for the time being but I'm not against reorganizing things differently if we ultimately think it makes more sense.